### PR TITLE
Add function for returning default os params

### PR DESCRIPTION
**Pull request by igchor**
To limit boilerplate in tests and user code.

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

File: src/memory_targets/memory_target_numa.c [hunk 1]
```diff
@@ -41,22 +41,6 @@ static umf_result_t numa_initialize(void *params, void **memTarget) {
...
 
-static const umf_os_memory_provider_params_t
-    UMF_OS_MEMORY_PROVIDER_PARAMS_DEFAULT = {
-        // Visibility & protection
-        .protection = UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
-        .visibility = UMF_VISIBILITY_PRIVATE,
-
-        // NUMA config
-        .nodemask = NULL,
-        .maxnode = 0, // TODO: numa_max_node/GetNumaHighestNodeNumber
```
Comment (lukaszstolarczuk) [inline src/memory_targets/memory_target_numa.c:52]: you could "save" these TODOs  (perhaps move them or file an issue for them), unless it's already been resolved 😉

File: test/provider_os_memory.cpp [hunk 2]
```diff
@@ -22,20 +22,6 @@ typedef enum purge_t {
...
-    /* .visibility = */ UMF_VISIBILITY_PRIVATE,
-
-    // NUMA config
-    /* .nodemask = */ NULL,
-    /* .maxnode = */ 0,
-    /* .numa_mode = */ UMF_NUMA_MODE_DEFAULT,
-    /* .numa_flags = */ 0,
-
-    // others
-    /* .traces = */ 1,
```
Comment (lukaszstolarczuk) [inline test/provider_os_memory.cpp:36]: traces set here to 1 (not sure if relevant, though). Please, confirm it's not just omitted by mistake

File: test/provider_os_memory_config.cpp [hunk 3]
```diff
@@ -19,22 +19,10 @@ static constexpr size_t allocSize = 4096;
...
-        /* .visibility = */ UMF_VISIBILITY_PRIVATE,
-
-        // NUMA config
-        /* .nodemask = */ NULL,
-        /* .maxnode = */ 0,
-        /* .numa_mode = */ UMF_NUMA_MODE_DEFAULT,
-        /* .numa_flags = */ 0,
-
-        // others
-        /* .traces = */ 1,
```
Comment (lukaszstolarczuk) [inline test/provider_os_memory_config.cpp:33]: .

File: src/memory_targets/memory_target_numa.c (hunk 1 above)
Comment (igchor) [reply src/memory_targets/memory_target_numa.c:52]: Those TODO are not longer relevant actually. This entire initialization code was just copied from somewhere with those TODOs but in this file, we specifically set maxnode to number of memory targets.

Comment (igchor) [reply src/memory_targets/memory_target_numa.c:52]: And in tests, we already have a mechanism for finding maxnode.

File: test/provider_os_memory.cpp (hunk 2 above)
Comment (igchor) [reply test/provider_os_memory.cpp:36]: It was intentional. Some ENV variable should control the traces and we should not hardcode it in tests to 1.

**Review by lukaszstolarczuk — COMMENTED** [review]
👍

**Comment by bratpiorka** [comment]
@lukaszstolarczuk @ldorau I need second approval before I merge these changes

**Review by ldorau — APPROVED** [review]

**Pull request merged**